### PR TITLE
chore: clean up single-used functions in Electra

### PR DESCRIPTION
**Pull request by ensi321**
`v1.5.0-alpha.7` includes several clean ups on the specs. Majority of them are removing definitions of single-used functions.

Remove:
- `getActiveBalance`
- `queueEntireBalanceAndResetValidator`
- Duplicate logic of `getMaxEffectiveBalance`
- `upgradeStateToElectraOriginal`

Waiting for alpha 7 feature PRs to be merged.

### CODE REVIEW

**Review by nflaig — COMMENTED** [review]
most of the changes here are duplicated in https://github.com/ChainSafe/lodestar/pull/7131

**Comment by ensi321** [comment]
> most of the changes here are duplicated in #7131

~~Yes this branch is based on `nc/delay-deposits`. Waiting for it to be merged before marking this as ready for review~~

Ready for review now

File: packages/state-transition/src/slot/upgradeStateToElectra.ts [hunk 1]
```diff
@@ -93,7 +90,23 @@ export function upgradeStateToElectra(stateDeneb: CachedBeaconStateDeneb): Cache
...
+    const validator = stateElectra.validators.get(validatorIndex);
+    validator.effectiveBalance = 0;
+    stateElectra.epochCtx.effectiveBalanceIncrementsSet(validatorIndex, 0);
+    validator.activationEligibilityEpoch = FAR_FUTURE_EPOCH;
+
+    const pendingDeposit = ssz.electra.PendingDeposit.toViewDU({
+      pubkey: validator.pubkey,
+      withdrawalCredentials: validator.withdrawalCredentials,
+      amount: balance,
+      signature: new Uint8Array(G2_POINT_AT_INFINITY),
```
Comment (nflaig) [inline packages/state-transition/src/slot/upgradeStateToElectra.ts]: this seems like an unnecessary conversion

we should probably convert everything to Uint8Array here as well
https://github.com/ChainSafe/lodestar/blob/unstable/packages/state-transition/src/constants/constants.ts#L1

**Review by nflaig — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.23.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.23.0) :tada:

**Pull request merged**